### PR TITLE
Remove integration tests for botocore with Python 2.7

### LINKED ISSUE #2423: Remove integration tests for Botocore with Python 2.7
Botocore dropped support for Python 2.7 in July so we don't have to do integration testing with Botocore+Python 2.7 on the 1.26.x branch any longer.

Reference: https://github.com/urllib3/urllib3/pull/2422

### PR DESCRIPTION
**Pull request by imkaka**
Closes #2423 

@sethmlarson Please let me know if anything else is needed here.

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -12,6 +12,11 @@ jobs:
       matrix:
         python-version: [2.7, 3.9]
         downstream: [botocore, requests]
+      exclude:
+        # excludes botocore in Python 2.7
+        - python-version: 2.7
+          downstream: botocore
+
```
Comment (pquentin) [inline .github/workflows/integration.yml:19]: This should be under `matrix`, the indentation is currently incorrect

Comment (imkaka) [reply .github/workflows/integration.yml]: @pquentin Thanks, fixed it please have a look.

**Review by pquentin — APPROVED** [review]
Thanks! This works now (tested in my own fork).

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**